### ISSUE TITLE
MLP can't learn XOR.

### ISSUE THREAD

**Issue by c19** [fchollet/keras#1295]
```python
from keras.models import Sequential
from keras.layers.core import Activation, Dense, Dropout
from keras.optimizers import SGD
import numpy as np
import theano
from theano import pp

X_test = X_train = np.array([[0, 0], [0, 1], [1, 0], [1, 1]])
Y_test = Y_train = np.array([0, 1, 1, 0])

model = Sequential()
model.add(Dense(2, input_shape=(2,)))
model.add(Activation('sigmoid'))
model.add(Dense(1))
model.add(Activation('sigmoid'))
model.compile(loss='binary_crossentropy', class_mode='binary', optimizer='sgd')

history = model.fit(X_train, Y_train, batch_size=4, verbose=1, shuffle=False, show_accuracy=True, nb_epoch=500)
score = model.evaluate(X_test, Y_test, show_accuracy=True)
print(score)
```
loss is dropping, but never get it right 
```python
model.predict(X_train)
array([[ 0.55641359],
       [ 0.51271898],
       [ 0.57324523],
       [ 0.52622819]])
```

I tried manually set the Ws and bs for each layer. and it got complete different result.
I checked the resulted network structure which is the same. I guess misunderstood something or something is really wrong. but I checked and experimented with different loss function. optimizers. had no clue :(

**Comment by fchollet** [comment]
See https://github.com/fchollet/keras/issues/112#issuecomment-101079731

**Comment by amit4111989** [comment]
from keras.models import Sequential
from keras.layers.core import Dense, Activation
from keras.optimizers import SGD
import numpy as np

np.random.seed(100)

model = Sequential()
model.add(Dense(2, input_dim=2))
model.add(Activation('sigmoid'))
model.add(Dense(1, input_dim=2))
model.add(Activation('sigmoid'))
X = np.array([[0,0],[0,1],[1,0],[1,1]], "float32")
y = np.array([[0],[1],[1],[0]], "float32")
sgd = SGD(lr=0.1, decay=1e-6, momentum=0.9, nesterov=True)
model.compile(loss='binary_crossentropy', optimizer=sgd)
model.fit(X, y, nb_epoch=1000, batch_size=1)

print(model.predict(X))
~
